specutils/io/default_loaders/ascii.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+import astropy.units as u
+from astropy.nddata import StdDevUncertainty
+from astropy.table import Table
+
+from specutils import Spectrum1D
+from specutils.io.registers import data_loader
+
+__all__ = ['ascii_identify', 'ascii_loader', 'ipac_identify', 'ipac_loader']
+
+
+def ascii_identify(origin, *args, **kwargs):
+    """Check if it's an ASCII file."""
+    name = os.path.basename(args[0])
+
+    if name.lower().split('.')[-1] in ['txt', 'ascii']:
+       return True
+
+    return False
+
+
+@data_loader(label="ASCII", identifier=ascii_identify, extensions=['txt', 'ascii'])
+def ascii_loader(file_name, **kwargs):
+    """
+    Load spectrum from ASCII file
+
+    Parameters
+    ----------
+    file_name: str
+        The path to the ASCII file
+
+    Returns
+    -------
+    data: Spectrum1D
+        The data.
+    """
+    table = Table.read(file_name, format='ascii')
+
+    flux = None
+    dispersion = None
+    uncertainty = None
+
+    # If there are more than two columns, assume that the first is wavelength,
```

Comment:
This seems very dangerous to me, as this is making assumptions that are very much not necessarily true and doing no validating.  This is the sort of reason why I was suggesting this "generic" reader not be in here...?

Reply:
This was added from spacetelescope/specviz#503. The question would be then: do we add _no_ default loaders that make generic assumptions (at some point) about the user's data, or do we build in _some_ basic assumptions?

ASCII is probably the most difficult, but we could parse the file and see if the astropy ascii loader can detect _any_ unit information, and if not, then perhaps raise an exception about ambiguity; but if so, assume certain defined columns are flux/spectral_axis/uncertainty?

Reply:
@eteq That was my suggestion to @SaOgaz, mostly motivated by having a way to make a guess that might not be terrible in the SpecViz context. But I'm okay with not having this loader as long as we have an wizard implemented in SpecViz that can help people with ascii or csv files, and we provide an example in the SpecUtils documentation of reading the file using astropy Table.read, adding units, and feeding it to Spectrum1D.

Reply:
To follow-through on this: in some slack discussion the conclusion was that it makes sense to allow the *generic* case but make it clear that this is an intentionally-limited "generic" that is roughly speaking the thing that the specviz wizard produces, but *no more* complicated than that.  That is, not put a lot of logic in the generic readers to figure out weird cases, but rather tell users that if it's not a simple column-to-spectrum1d attribute mapping, they should write a custom reader.  So sort of a middle ground (yay compromise!)